engine/crates/engine-v2/src/operation/bind/selection_set.rs

```diff
@@ -27,7 +28,11 @@ pub(super) struct SelectionSetBinder<'schema, 'parsed, 'binder> {
     #[allow(clippy::type_complexity)]
     fields: HashMap<
         (SafeResponseKey, FieldDefinitionId),
-        (QueryPosition, Vec<&'parsed Positioned<engine_parser::types::Field>>),
+        (
+            QueryPosition,
+            Vec<&'parsed Positioned<engine_parser::types::Field>>,
+            Vec<QueryModifierIndicator>,
+        ),
```

Comment:
I may be misunderstanding this, but what happens if the same field appears under completely separate `QueryModifierIndicator`s?  Can this handle that?

Reply:
I wrote a failing test that suggests it can't:

```
#[test]
fn skip_include_again() {
    let response = runtime().block_on(async move {
        let engine = Engine::builder().with_subgraph(FakeGithubSchema).build().await;

        engine
            .post(
                r#"
                    query PullRequests($a: Boolean!, $b: Boolean!) {
                        pullRequestsAndIssues(filter: { search: "1" }) {
                            ... on PullRequest @include(if: $a) {
                                title
                            }
                            ... on PullRequest @include(if: $b) {
                                title
                            }
                        }
                    }
                    "#,
            )
            .variables(json!({"a": true, "b": false}))
            .await
    });

    insta::assert_json_snapshot!(response, @r###"
    {
      "data": {
        "pullRequestsAndIssues": [
          {
            "title": "Creating the thing"
          },
          {
            "title": "Some bot PR"
          },
          {}
        ]
      }
    }
    "###);
}
```

Reply:
@obmarg is the snapshot what you expect or the result?

Reply:
The snapshot is what should be returned.  But it just returns a list of empty objects.

Reply:
You can copy and paste that into your test file and it should be runnable if you want to see.

Reply:
Theoretically this is the same as 

```
{
  title @include(if: true)
  title @include(if: false)
 }
 ```
 which are merged to one field (as in your example) - When discussing with @Finistere we said that if any directive sets the field to skip that should affect all directives (which is the behavior when you set multiple directives or both include and skip to the same field) - So as far as I can tell it's working as expected

Reply:
The spec doesn't really define the former, it does define the latter (multiple on one field rather than multiple instances of one field)

Reply:
In general the behavior is that a single true (or false for include) for a field overrides anything else

Reply:
(but it does consider all rules for the same field if that's what you're asking, or at least should unless I missed something)

Reply:
That is not how skip and include are specified as working though.

Reply:
e.g. 

```
{ 
  title @skip(if: false) @include(if:true)
 }
 ```
 and so forth will look at all directives and omit the field only if one of them is the "off" value for the relevant directive

Reply:
@obmarg how do you mean?

>Neither @skip nor @include has precedence over the other. In the case that both the @skip and @include directives are provided on the same field or fragment, it must be queried only if the @skip condition is false and the @include condition is true. Stated conversely, the field or fragment must not be queried if either the @skip condition is true or the @include condition is false.

Regarding merging multiple instances of the same field it doesn't really specify anything, and that's how our engine works (as in there isn't multiple instances of the same field when the query is planned, they're merged)

Reply:
The spec is pretty clear that you look at each individual selection individually and decide whether or not to include that selection based on the directive that is applied to it.  See step 3.a & 3.b here: http://spec.graphql.org/October2021/#CollectFields()

Reply:
Actually now that I reread our conversation @Finistere I think you might've meant this as well. I'd misread since I'd been going by the behavior of multiple instances on the same field. I'll look into this tomorrow

Reply:
Good catch @obmarg

Reply:
Fixed